controllers/dnspolicy_controller.go

```diff
@@ -199,5 +202,35 @@ func (r *DNSPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				return gatewayEventMapper.MapToPolicy(ctx, object, &v1alpha1.DNSPolicy{})
 			}),
 		)
+	r.RegisterEventHandler(&DNSMetricsEventHandler{})
 	return ctrlr.Complete(r)
 }
+
+const (
+	dnsRecordNameLabel      = "dns_record_name"
+	dnsRecordNamespaceLabel = "dns_record_namespace"
+	dnsRecordEventTypeLabel = "dns_record_event_type"
+	dnsPolicyNameLabel      = "dns_policy_name"
+	dnsPolicyNamespaceLabel = "dns_policy_namespace"
+	dnsPolicyReadyLabel     = "dns_policy_ready"
+)
+
+var (
+	dnsRecordEvent = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "dns_record_event",
+			Help: "DNS Record events",
+		},
+		[]string{dnsRecordNameLabel, dnsRecordNamespaceLabel, dnsRecordEventTypeLabel})
+	dnsPolicyReady = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "dns_policy_ready",
```

Comment:
When dealing with the state of resources, I've found the patterns in the kube-state-metrics project to be quite good.
Check out the `kube_pod_status_ready` metrics (and variations) at https://github.com/kubernetes/kube-state-metrics/blob/main/docs/metrics/workload/pod-metrics.md

Reply:
I'm not sure I'm following how to reconcile this with the above conversation. I was under the impression the current approach was inferior as it has the status as a label (so then, what is the gauge gauging), but the kube_pod_status_ready has the status as a label. Am I missing something? 

Or, maybe an alternative question: When does the `kube_pod_status_ready` gauge increment?

Reply:
Maybe an example set of metrics will help understand the pattern used by kube-state-metrics.
Here's a set for 1 pod:

```
kube_pod_status_ready{condition="false", namespace="demo-petstore", pod="petstore-6f58b6dd67-b2ncj"} 0
kube_pod_status_ready{condition="true", namespace="demo-petstore", pod="petstore-6f58b6dd67-b2ncj"} 1
kube_pod_status_ready{condition="unknown", namespace="demo-petstore", pod="petstore-6f58b6dd67-b2ncj"} 0
```
From this current state, we can say that the `petstore-6f58b6dd67-b2ncj` pod Ready condition is currently 'true'.
A time series exists for each condition, and the guage value changes to 0 to 1 depending on if that ready condition is the active one.

Reply:
oh I see, so when it goes:
```
ready: false
- emit with label condition false: 1
- emit with label condition true: 0
- emit with label condition unknown: 0
ready: true
- emit with label condition false: 0
- emit with label condition true: 1
- emit with label condition unknown: 0
ready: absent
- emit with label condition false: 0
- emit with label condition true: 0
- emit with label condition unknown: 1
```